telemetry/data_pipeline_mvp/grab_from_serial.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+"""
+objective - grab data from serial, parse, pass into redis timeseries 
+
+"""  
+
+import serial
+import time
+from redistimeseries.client import Client
+
+# figure out way to automatically detect port from python? 
+
+
+
+class Data_parser(object):
+    def __init__(self):
+        self.port_name = "/dev/ttyUSB0"
+        self.baudrate = 9600
+        self.read_frequency = 20 # in hz  
+        self.data_channels = ['arduino_data']
+
+        # init serial connection
+        self.ser = serial.Serial(self.port_name, self.baudrate)  
```

Comment:
i would consider maybe making a `DataStream` object that Serial can fit into? the reason for this would be so that we can easily unit test this code without needing an actual serial device (if we simulate one)

What this might look like is a seperate `DataStream` object that gets passed into this class that has a function `get_data` (or something). That class is then extended to a `SerialDataStream` and a `SimulatedDataStream`, where we can inject our own `get_data` method that feeds placeholder data.

Reply:
The idea is to make our code *as generic as possible*. It would be great if we had an alternate data input (for example, a log file maybe?) that we can stream directly back into this interface and have the whole system work smoothly

Reply:
I'd also consider whether we might want to instead interface with the serial data stream in C and pipe the results into another program that accepts input from any stream. It's the Unix philosophy that everything should have a single job and be piped together. There are also certain elements we might want to rewrite in C if we find it to be faster.

Not necessarily for MVP, but for a more refined draft.